test/boost_ext/CompressedStorageComparison.hpp

```diff
@@ -0,0 +1,107 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_BOOST_TEST_COMPRESSED_STORAGE_COMPARISON_HPP
+#define ARBORX_BOOST_TEST_COMPRESSED_STORAGE_COMPARISON_HPP
+
+#include <boost/test/tools/detail/print_helper.hpp>
+
+#include <iosfwd>
+#include <set>
+
+template <typename Offsets, typename Values>
+struct CompressedStorage
+{
+  Offsets offsets;
+  Values values;
+  using value_type = typename Values::value_type;
+  using index_type = typename Offsets::value_type;
+  struct ConstForwardIterator
+  {
+    index_type i;
+    CompressedStorage const *p;
+    using value_type = std::multiset<CompressedStorage::value_type>;
```

Comment:
Is is multiset only because we could get it in an error? In normal situations, we should get uniques, right?

Reply:
Right.